tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java

```diff
@@ -356,15 +360,18 @@ private void setBreakpoints(List<BallerinaTestDebugPoint> breakPoints) throws Ba
                     .stackTrace(stackTraceArgs);
             StackFrame[] stackFrames = stackTraceResp.getStackFrames();
             if (stackFrames.length == 0) {
-                return new Variable[0];
+                return variables;
+            }
+            if (scope == VariableScope.LOCAL) {
+                scopeArgs.setFrameId(stackFrames[0].getId());
+            } else {
+                scopeArgs.setFrameId(-stackFrames[0].getId());
             }
-
-            scopeArgs.setFrameId(stackFrames[0].getId());
             ScopesResponse scopesResp = DebugHitListener.connector.getRequestManager().scopes(scopeArgs);
             variableArgs.setVariablesReference(scopesResp.getScopes()[0].getVariablesReference());
             VariablesResponse variableResp = DebugHitListener.connector.getRequestManager().variables(variableArgs);
-            return variableResp.getVariables();
-
+            Arrays.stream(variableResp.getVariables()).forEach(variable -> variables.put(variable.getName(), variable));
```

Comment:
As a practice, let's try to get each of the stream operations to a new line

Reply:
Kept it inline as there's only one stream operation in here. changed with db9e03c.